pkg/config/datafileprojectconfig/optimizely_config.go

```diff
@@ -0,0 +1,145 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package datafileprojectconfig //
+package datafileprojectconfig
```

Comment:
This is the wrong package to put it in. The `OptimizelyConfig` shouldn't be tightly coupled to the concept of a `datafile`. If it's the package name confusion that we care about I'd rather us just do `pkg/config/optimizely_config.go` and have the linter ignore it heh. I understand that the package name kind of collides with the file and class names but that is more syntactically correct for us